packages/app/src/cli/services/context.ts

```diff
@@ -430,7 +430,7 @@ export async function ensureDeployContext(options: DeployContextOptions): Promis
   const {reset, force, noRelease} = options
   let developerPlatformClient = options.developerPlatformClient
   // do the link here
-  const [remoteApp] = await fetchAppAndIdentifiers(options, developerPlatformClient, true, true)
+  const [remoteApp] = await fetchAppAndIdentifiers(options, developerPlatformClient, true, !options.apiKey)
```

Comment:
Is this logic something we only want to apply for deploy? For example, `release` also works the same way. `fetchAppAndIdentifiers ` has access to `options` so in theory we could do this logic inside of `fetchAppAndIdentifiers` if we wanted to apply this business logic to anyone who is trying to use this method.

No strong opinions...just asking...